tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/ant/Task.java

```diff
@@ -0,0 +1,25 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.icu.tool.cldrtoicu.ant;
+
+public class Task {
```

Comment:
It looks like `Task` should be an interface with default methods. I don't think that stops you from having inner classes inside it, so such a change would be minimal.

Reply:
That is what I initially did.

But there are are many classes extending Task, not just the top 3: \
`LocaleIds`, `Directory`, `ForcedAlias`, `AltPath` in `ConvertIcuDataTask` \
`Retain`, `Dir` in `CleanOutputDirectoryTask`

Some implement init, some implement exec, some implement both.

So to make the Task an interface I would need to either add some empty implementations in each of these classes, or add default implementations in the interface.

First option added too much noise (my goal was to keep things as close to the original).

Second option means that instead of:
```java
class Task {
   log() { print }
   init() {}
   exec() {}
}
```
I would have:
```java
interface Task {
   default log() { print }
   default init() {}
   default exec() {}
}
```

It does not look like an improvement.

Reply:
You're right, not worth doing.  An interface's behaviors should be implemented by all participants. If that's not the case, then maybe it's a sign of 2 different behaviors, each with their own interface. This Task design is inherited from the Ant-based way of doing things, and since it works, it's better to leave as is.